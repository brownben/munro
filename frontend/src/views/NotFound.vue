<!--
  Not Found

  Displayed when page is not found by the router, or data is not found from API
-->

<template>
  <div>
    <vue-headful
      title="Munro - Not Found"
      :head="{
        'meta': {name: 'robots', content:'noindex'},
      }"
    />
    <h1>Sorry We Can't Find What You Are Looking For</h1>
    <p>Please check the URL and Try Again</p>
    <button @click="$router.go(-1)">Go Back</button>
    <router-link to="/" class="button">Return Home</router-link>
  </div>
</template>

<style lang="stylus" scoped>
@import '../assets/styles/helpers.styl'
@import '../assets/styles/inputs.styl'

p
  padding: 0.5rem 0

button
  margin-top: 0.25rem
  margin-right: 1rem
</style>
