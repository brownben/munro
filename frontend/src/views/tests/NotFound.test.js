/*
  Not Found View Unit Tests
*/

import { mount } from '@vue/test-utils'
import NotFound from '@/views/NotFound'

test('Is a Vue Instance', () => {
  const wrapper = mount(NotFound)
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = mount(NotFound)
  expect(wrapper.element).toMatchSnapshot()
  expect(wrapper.find('h1').text()).toBe('Sorry We Can\'t Find What You Are Looking For')
  expect(wrapper.findAll('button').length).toBe(2)
})
