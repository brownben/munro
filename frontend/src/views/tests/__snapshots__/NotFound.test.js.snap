// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<div>
  <vue-headful-stub
    head="[object Object]"
    title="Munro - Not Found"
  />
   
  <h1>
    Sorry We Can't Find What You Are Looking For
  </h1>
   
  <p>
    Please check the URL and Try Again
  </p>
   
  <button>
    Go Back
  </button>
   
  <router-link-stub
    class="button"
    to="/"
  >
    Return Home
  </router-link-stub>
</div>
`;
